riscv_analysis/src/cfg/function.rs

```diff
@@ -55,14 +72,45 @@ impl Function {
 
     #[must_use]
     pub fn returns(&self) -> HashSet<Register> {
-        self.exit.live_in().intersection_c(&RegSets::ret())
+        self.exit().live_in().intersection_c(&RegSets::ret())
+    }
+
+    /// Insert the set of registers used by this function.
+    pub fn insert_defs(&self, defs: HashSet<Register>) {
```

Comment:
This function should be called `set` or `reset` instead of `insert`. `insert` implies we are combining the two sets, not replacing the whole sets. Ditto for the other `insert` functions that replace the underlying function.